src/dbnode/client/config.go

```diff
@@ -101,10 +101,18 @@ type Configuration struct {
 	// operations the client will attempt to buffer for each host before issuing an RPC.
 	TargetHostQueueFlushSize *int `yaml:"targetHostQueueFlushSize"`
 
+	// AsyncTargetHostQueueFlushSize sets the target size for an async host queue flush. This controls how many
+	// operations the client will attempt to buffer for each host before issuing an RPC.
+	AsyncTargetHostQueueFlushSize *int `yaml:"asyncTargetHostQueueFlushSize"`
+
 	// HostQueueFlushInterval sets the interval at which the m3db client will flush the queue for a
 	// given host regardless of the number of batched operations.
 	HostQueueFlushInterval *time.Duration `yaml:"hostQueueFlushInterval"`
 
+	// AsyncHostQueueFlushInterval sets the interval at which the m3db client will flush the queue for a
+	// given async host regardless of the number of batched operations.
+	AsyncHostQueueFlushInterval *time.Duration `yaml:"asyncHostQueueFlushInterval"`
+
```

Comment:
Could this be moved into `environment.Config` so that we can override on a per-cluster basis? It might be a little brittle for us to have a single override for all async clusters... even though it fits our current use-case, it would be a breaking change for us to move to per-cluster overrides in the future.